### PR TITLE
feat: migration astro

### PR DESCRIPTION
**Pull request by sileht**
Fixes MRGFY-2995
Fixes MRGFY-2984

### CODE REVIEW

**Review by Syffe — DISMISSED** [review]

**Comment by jd** [comment]
Can we have more items per line here like we have currently:
CleanShot 2024-01-03 at 11 06 48

**Review by jd — CHANGES_REQUESTED** [review]
See comment

**Comment by jd** [comment]
CleanShot 2024-01-03 at 11 08 07

**Comment by jd** [comment]
CleanShot 2024-01-03 at 11 11 24

**Comment by jd** [comment]
CleanShot 2024-01-03 at 11 12 04

**Comment by jd** [comment]
CleanShot 2024-01-03 at 11 13 44

File: .github/workflows/ci-docs.yaml [hunk 1]
```diff
@@ -26,9 +26,11 @@ jobs:
           DEST_REF: refs/heads/preview/${{ github.event.pull_request.number }}
         run: |
           if gh api --silent "/repos/Mergifyio/docs/git/${DEST_REF}"; then
-            gh api -X POST "/repos/Mergifyio/docs/git" -f "sha=${HEAD_SHA}" -f "refs=$DEST_REF"
-          else
+            echo "Updating $DEST_REF"
```
Comment (jd) [inline .github/workflows/ci-docs.yaml:29]: Should this be in another PR?

**Comment by jd** [comment]
CleanShot 2024-01-03 at 11 18 36

File: src/components/HeadCommon.astro [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		range.setEnd(node, 1);
+		const selection = document.getSelection();
+		if (selection) {
+			selection.removeAllRanges();
+			selection.addRange(range);
+		}
+	});
+</script>
+<!-- Fathom - beautiful, simple website analytics -->
+<script src="https://cdn.usefathom.com/script.js" data-site="EZBHTSIG" defer></script>
```
Comment (jd) [inline src/components/HeadCommon.astro:65]:
```suggestion
```

**Review by jd — DISMISSED** [review]

**Review by jd — COMMENTED** [review]
(Is plausible set up also?)

**Comment by jd** [comment]
Could we get back the footer also?

**Comment by AlexandreGaubert** [comment]
> (Is plausible set up also?)

I've just added the plausible script to the `head`

File: .github/workflows/ci-docs.yaml (hunk 1 above)
Comment (AlexandreGaubert) [reply .github/workflows/ci-docs.yaml:29]: @sileht This one is for you I think

**Comment by AlexandreGaubert** [comment]
@jd I've made the corrections, but for the code snippet bg color I've switched to a dark shade of the mergify blue (which is **slightly** lighter than the previous one), tell me if it suits you.

**Review by sileht — COMMENTED** [review]
I can't approve myself, but it looks good for me!

**Comment by jd** [comment]
Just one last thing: can we use something more blue and less purple here too on links:
CleanShot 2024-01-04 at 16 25 07

File: .github/workflows/ci-docs.yaml (hunk 1 above)
Comment (sileht) [reply .github/workflows/ci-docs.yaml:29]: We need to rebase the branch to fix that one, because this change is already in main.

Comment (sileht) [reply .github/workflows/ci-docs.yaml:29]: @AlexandreGaubert feel free to rebase the branch and force-push when you fix the last @jd  comment.

**Comment by AlexandreGaubert** [comment]
@jd @sileht Should be good now 🙂

File: static/mergify-count-active-users.py [hunk 3]
```diff
@@ -1,175 +0,0 @@
-#!/usr/bin/env python3
-# This scripts counts the number of active users for repositories over the last 30 days.
```
Comment (jd) [inline static/mergify-count-active-users.py:2]: Where is this file now?

Comment (AlexandreGaubert) [reply static/mergify-count-active-users.py]: Missed it, it's here now

File: src/content.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	isRecipeEntry,
+	isTutorialEntry,
+} from './content/config';
+
+const isKoreanEntry = createIsLangEntry('ko');
+
+export const allPages = await getCollection('docs', (entry) => {
+	if (import.meta.env.PUBLIC_TWO_LANG) {
+		// Build for two languages only to speed up Astro's smoke tests
+		return isEnglishEntry(entry) || isKoreanEntry(entry);
```
Comment (jd) [inline src/content.ts:14]: ?

File: src/content.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+	} else {
+		return true;
+	}
+});
+
+export const tutorialPages = allPages.filter(isTutorialEntry);
+export const recipePages = allPages.filter(isRecipeEntry);
+export const englishPages = allPages.filter(isEnglishEntry);
```
Comment (jd) [inline src/content.ts:22]: ?

File: src/content/config.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+	service: z.string(),
+});
+
+export const integrationSchema = baseSchema.extend({
+	type: z.literal('integration'),
+	title: z
+		.string()
+		.refine(
+			(title) => title.startsWith('@astrojs/'),
+			'"title" must start with "@astrojs/" for integration docs.'
```
Comment (jd) [inline src/content/config.ts:35]: ?

**Review by jd — CHANGES_REQUESTED** [review]

**Review by jd — COMMENTED** [review]
src/content/config.ts is empty. remove it?

**Comment by AlexandreGaubert** [comment]
> src/content/config.ts is empty. remove it?

@jd Removed

File: src/content/docs/integrations/ci-further.mdx [hunk 7]
```diff
@@ -1,3 +1,7 @@
+---
+title: 'Ci further'
+---
```
Comment (jd) [inline src/content/docs/integrations/ci-further.mdx:3]: Is this mandatory? Should we prefix the file with `_` to avoid this?
(the page is only used for include)

Comment (AlexandreGaubert) [reply src/content/docs/integrations/ci-further.mdx]: Done

File: src/env.d.ts [hunk 8]
```diff
@@ -0,0 +1,2 @@
+/// <reference path="../.astro/types.d.ts" />
```
Comment (jd) [inline src/env.d.ts:1]: delete this file?

File: src/util/generateToc.test.ts [hunk 9]
```diff
@@ -0,0 +1,80 @@
+import { describe, expect, test } from 'vitest';
```
Comment (jd) [inline src/util/generateToc.test.ts:1]: Is this run in CI?

File: src/env.d.ts (hunk 8 above)
Comment (AlexandreGaubert) [reply src/env.d.ts:1]: It reference a typing file generated on build (dev or prod) which contains typing specific to astro structure. This file is not commited, maybe we could commit it, it was done like that in the template.

File: src/util/generateToc.test.ts (hunk 9 above)
Comment (AlexandreGaubert) [reply src/util/generateToc.test.ts:1]: No, i'm adding it

Comment (AlexandreGaubert) [reply src/util/generateToc.test.ts:1]: Done

File: src/env.d.ts (hunk 8 above)
Comment (jd) [reply src/env.d.ts:1]: but is `env.d.ts` file needed?

**Review by jd — APPROVED** [review]

File: src/env.d.ts (hunk 8 above)
Comment (AlexandreGaubert) [reply src/env.d.ts:1]: Yes, it reference both `.astro/types.d.ts` and `astro/client`, and typescript is integrating it for typing

**Review by AlexandreGaubert — APPROVED** [review]

**Review by jd — APPROVED** [review]

**Review by AlexandreGaubert — APPROVED** [review]
